packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -246,6 +251,12 @@ class AdaptiveScaffold extends StatefulWidget {
   /// [Breakpoint].
   final double extendedNavigationRailWidth;
 
+  /// Used to map NavigationDestination to NavigationRailDestination.
+  final NavigationRailDestination Function(
```

Comment:
Consider extracting the method signature into a typedef. https://github.com/flutter/packages/blob/5cc0a016019451e26b2fecf31923894e371b6335/packages/go_router/lib/src/builder.dart#L19-L23 for example.